client/components/app-promo/index.jsx

```diff
@@ -80,13 +76,13 @@ export default React.createClass( {
 			return null;
 		}
 
-		const promo_link = 'https://apps.wordpress.com/desktop/?ref=promo_' + this.props.location + '_' + this.state.promo_item.promo_code;
+		const promo_link = getPromoLink( this.props.location, this.state.promo_item.promo_code );
```

Comment:
similar comment as with above, we can pass this in as a prop and then we won't have to mock it…

```js
const { getPromoLink } = this.props;
const { promo_item } = this.state;

const promoLink = getPromoLink( promo_item.promo_code );
```

let's also try and use `camelCase` in Calypso variable names